File: day-02/workspace/java-oop/src/main/java/com/targetindia/model/Circle.java
package com.targetindia.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Circle implements GeometricShape{
    private double radius;

    @Override
    public double calculateArea() {
        return PI*radius*radius;
    }

    @Override
    public String getShapeName() {
        return "CIRCLE";
    }
}

File: day-13-advanced-kotlin-concepts/workspace/kotlin-interfaces/src/main/kotlin/com/targetindia/model/Circle.kt
package com.targetindia.model

class Circle : IGeometricShape {

    var x: Double = 0.0
    var y: Double = 0.0
    var radius: Double = 0.0

    constructor()

    constructor(radius: Double) {
        this.radius = radius
    }

    override fun getName(): String {
        return "Circle";
    }

    override fun getArea(): Double {
        return kotlin.math.PI * radius * radius
    }
}